pkg/devicetwin/devicetwin_test.go

```diff
@@ -0,0 +1,194 @@
+	querySeterMock = beego.NewMockQuerySeter(mockCtrl)
+	fakeModule = beehive.NewMockModule(mockCtrl)
+	dbm.DBAccess = ormerMock
+}
+
+// registerFakeModule is fuction to register all mock modules used.
+func registerFakeModule() {
+	fakeModule.EXPECT().Name().Return(TestModule).Times(3)
+	core.Register(fakeModule)
+	mainContext = context.GetContext(context.MsgCtxTypeChannel)
+	mainContext.AddModule(TestModule)
+}
+
+// TestName is function to test Name().
+func TestName(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+	}{
+		{
+			name: "DeviceTwinNametest",
+			want: "twin",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dt := &DeviceTwin{}
+			if got := dt.Name(); got != tt.want {
+				t.Errorf("DeviceTwin.Name() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TestGroup is function to test Group().
+func TestGroup(t *testing.T) {
+	tests := []struct {
+		name string
+		want string
+	}{
+		{
+			name: "DeviceTwinGroupTest",
+			want: "twin",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dt := &DeviceTwin{}
+			if got := dt.Group(); got != tt.want {
+				t.Errorf("DeviceTwin.Group() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TestStart is function to test Start().
+func TestStart(t *testing.T) {
+	initMocks(t)
+	registerFakeModule()
+	core.Register(&DeviceTwin{})
+	dt := DeviceTwin{}
+	mainContext.AddModule(dt.Name())
+	mainContext.AddModuleGroup(dt.Name(), dt.Group())
+	dt.context = mainContext
+	controller, err := InitDTController(mainContext)
+	if err != nil {
+		t.Errorf("Start device twin failed, due to %v", err)
+		return
+	}
+	dt.dtcontroller = controller
+	ormerMock.EXPECT().QueryTable(gomock.Any()).Return(querySeterMock).Times(1)
+	querySeterMock.EXPECT().All(gomock.Any()).Return(int64(1), nil).Times(1)
+	go dt.Start(mainContext)
+	time.Sleep(1 * time.Millisecond)
+	// Sending a message from devicetwin module to the created fake module(TestModule) to check context is initialized properly.
+	dt.context.Send(TestModule, test)
+	_, err = mainContext.Receive(TestModule)
+	t.Run("MessagePingTest", func(t *testing.T) {
+		if err != nil {
+			t.Errorf("Error while receiving message: %v", err)
+			return
+		}
+	})
+	//Checking whether Mem,Twin,Device and Comm modules are registered and started successfully.
+	tests := []struct {
+		name       string
+		moduleName string
+	}{
+		{
+			name:       "MemModuleHealthCheck",
+			moduleName: dtcommon.MemModule,
+		},
+		{
+			name:       "TwinModuleHealthCheck",
+			moduleName: dtcommon.TwinModule,
+		},
+		{
+			name:       "DeviceModuleHealthCheck",
+			moduleName: dtcommon.DeviceModule,
+		},
+		{
+			name:       "CommModuleHealthCheck",
+			moduleName: dtcommon.CommModule,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			moduleCheck := false
+			for _, module := range dt.dtcontroller.DTModules {
+				if test.moduleName == module.Name {
```

Comment:
There should be a ```go moduleCheck = false``` over here as we should overwrite the true value of previous iteration